include/zephyr/sys/util_internal.h

```diff
@@ -198,4 +198,7 @@
 /* Used by UTIL_X2 */
 #include "util_internal_util_x2.h"
 
+/* Simple math operations such as UTIL_ADD, UTIL_LSH, etc. use this */
+#include "util_internal_math.h"
```

Comment:
Honestly, I didn't feel any difference, although, that's probably just the luxury of modern computers... We could make it configurable with something like CONFIG_USE_PREPROCESS_MATH=y.

Regarding whether we want to go down that path. I mean, it's useful and since we're using mostly C, I dare argue that maintaining this isn't wasted effort. Especially considering the use cases together with the device-tree. There were many occasions where I wanted to use some simple math operations together with a comparison within the COND_CODE_ macros. It's not that I didn't any way around it. You simply have to use #if #else #endif and define the same macro with different values...